mir_eval/io.py

```diff
@@ -70,6 +75,12 @@ def load_delimited(filename, converters, delimiter=r'\s+'):
     # Create re object for splitting lines
     splitter = re.compile(delimiter)
 
+    # And one for comments
+    if comment is None:
+        commenter = None
```

Comment:
I don't think you need this condition, right? The only place `commenter` is used is in the line 
```Python
if comment is not None and commenter.match(line):
```
so if `comment is None` the `commenter` variable will never be encountered, and you can just do 
```Python
if comment is not None:
    commenter = re.compile('^{}'.format(comment))
```
here. Unless I am misunderstanding. (same note about line 629)

Reply:
That's true, but linters will be annoyed.

Reply:
Really? I thought the linters were smarter than that.

Reply:
Some are, not all.  I had originally written it as a short-circuit the way you suggest, but it's a bit more readable making things explicit IMO.